src/main/java/com/redhat/cloud/notifications/auth/RhServiceAccountIdentity.java

```diff
@@ -0,0 +1,43 @@
+package com.redhat.cloud.notifications.auth;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.quarkus.logging.Log;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RhServiceAccountIdentity extends Identity {
+
+    @JsonProperty("org_id")
+    private String orgId;
+
+    @JsonProperty("service_account")
+    private ServiceAccount serviceAccount;
+
+    public ServiceAccount getServiceAccount() {
+        return serviceAccount;
+    }
+
+    @Override
+    public String getSubject() {
+        return getServiceAccount().getUsername();
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class ServiceAccount {
```

Comment:
Should we check the scope of the service account? Maybe we should reject the request if that scope is not `api.notifications`. I'm not sure.

Reply:
I'm not sure if the scope matters here, let see how turnpike will be updated to restrict access to some routes, like it does with X509.